enumerables.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    if is_a? Hash
+      length.times { |element| yield(keys[element], self[keys[element]]) }
+    else
+      length.times { |element| yield(self[element]) }
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    length.times { |element| yield(self[element], element) }
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |element| new_array.push(element) if yield(element) }
+    new_array
+  end
+
+  def my_all?(*par)
+    condition = true
+    if !par[0].nil?
+      my_each { |element| condition = false unless par[0] === element } # rubocop:disable Style/CaseEquality
+    elsif !block_given?
+      my_each { |element| condition = false unless element }
+    else
+      my_each { |element| condition = false unless yield(element) }
+    end
+    condition
+  end
+
+  def my_any?(*par)
+    condition = false
+    if !par[0].nil?
+      my_each { |element| condition = true if par[0] === element } # rubocop:disable Style/CaseEquality
+    elsif !block_given?
+      my_each { |element| condition = true if element }
+    else
+      my_each { |element| condition = true if yield(element) }
+    end
+    condition
+  end
+
+  def my_none?(*par)
+    condition = true
+    if !par[0].nil?
+      my_each { |element| condition = false if par[0] === element } # rubocop:disable Style/CaseEquality
+    elsif !block_given?
+      my_each { |element| condition = false if element }
+    else
+      my_each { |element| condition = false if yield(element) }
+    end
+    condition
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:
```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?

Reply:
Fixed! Thank you for help @euqeme!